src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.tsx

```diff
@@ -43,7 +43,7 @@ interface DataSourceSelectableState {
   dataSourceOptions: SelectedDataSourceOption[];
   isPopoverOpen: boolean;
   selectedOption?: SelectedDataSourceOption[];
-  defaultDataSource: string | null;
```

Comment:
I thought we need this union type since we set it like `const defaultDataSource = this.props.uiSettings?.get('defaultDataSource', null) ?? null;`